wgpu-hal/examples/raw-gles.rs

```diff
@@ -10,61 +10,197 @@
 
 extern crate wgpu_hal as hal;
 
-#[cfg(not(any(windows, target_arch = "wasm32")))]
+#[cfg(not(any(windows, target_arch = "wasm32", target_os = "ios")))]
 fn main() {
+    use std::{ffi::CString, num::NonZeroU32};
+
+    use glutin::{
+        config::GlConfig as _,
+        context::{NotCurrentGlContext, PossiblyCurrentGlContext as _},
+        display::{GetGlDisplay, GlDisplay},
+        surface::GlSurface as _,
+    };
+    use glutin_winit::GlWindow as _;
+    use rwh_05::HasRawWindowHandle as _;
+
     env_logger::init();
     println!("Initializing external GL context");
 
-    let event_loop = glutin::event_loop::EventLoop::new();
-    let window_builder = glutin::window::WindowBuilder::new();
-    let gl_context = unsafe {
-        glutin::ContextBuilder::new()
-            .with_gl(glutin::GlRequest::Specific(glutin::Api::OpenGlEs, (3, 0)))
-            .build_windowed(window_builder, &event_loop)
-            .unwrap()
-            .make_current()
-            .unwrap()
-    };
-    let inner_size = gl_context.window().inner_size();
+    let event_loop = winit::event_loop::EventLoop::new().unwrap();
+    // Only Windows requires the window to be present before creating the display.
+    // Other platforms don't really need one.
+    let window_builder = cfg!(windows).then(|| {
+        winit::window::WindowBuilder::new()
+            .with_title("WGPU raw GLES example (press Escape to exit)")
+    });
```

Comment:
Do we need this if Windows isn't supported in the example?

Reply:
I'm adding that in #6152, fine to rip this out temporarily in this PR?

Reply:
Ah yup, no worries.